### PR TITLE
feat: automated fetching and verification of lens profile

### PR DESCRIPTION
**Pull request by spantheslayer**
## What does this PR do?
Automates Fetching and Verification of Lens Profile from  following

## Related issues
No

Fixes # (issue)
https://github.com/lensterxyz/lenster/issues/3347

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Explanation of the changes
It adds automation which takes lens profile following and adds them to verified list

### CODE REVIEW

File: packages/data/package.json [hunk 1]
```diff
@@ -15,5 +15,8 @@
     "@lenster/config": "*",
     "@types/node": "^20.4.4",
     "typescript": "^5.1.6"
+  },
+  "dependencies": {
+    "node-fetch": "^3.3.1"
```
Comment (bigint) [inline packages/data/package.json:20]: By default there is fetch right, no need of this, try removing and run

File: packages/data/verify.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+        // Load the verified.ts file
+        fs.readFile('./verified.ts', 'utf8', (err, tsData) => {
+          if (err) {
+            console.error('Error reading the verified.ts file:', err);
+            return;
+          }
+
+          const regex =
+            /export const mainnetVerified = \[[\S\s]*?...mainnetLensTeamMembers,[\S\s]*?...mainnetStaffs\n];/;
```
Comment (bigint) [inline packages/data/verify.js:87]: We don't wanna spread those two arrays, cuz by default we follow all people need to be verified

**Review by bigint — CHANGES_REQUESTED** [review]
Do we really need resp.json instead is it possible to store values on local variable also really we don't want to save the username in comment just array of profile Id is enough

File: packages/data/package.json (hunk 1 above)
Comment (spantheslayer) [reply packages/data/package.json]: removed the node-fetch pkg and enabled local variable to store the values

File: packages/data/verify.js (hunk 2 above)
Comment (spantheslayer) [reply packages/data/verify.js]: I am doing this to append the new data into the verified array before getting the unique data.

either way I have to take it to consideration even if I want to do it with existing data that I get after parsing the response, it's just would have to do the same thing post I get the unique data.

**Comment by spantheslayer** [comment]
@bigint please check, I have updated the code as requested.

File: packages/data/verify.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        }
+        pageInfo {
+          prev
+          next
+        }
+      }
+    }
+  `;
+
+  const url = 'https://api.lens.dev/';
```
Comment (bigint) [inline packages/data/verify.js:24]:
```suggestion
  const url = 'https://api.lens.dev';
```

File: packages/data/verify.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  const allResponses = [];
+
+  let cursor = null;
+  while (true) {
+    // const fetch = await import('node-fetch');
+    const response = await fetch(url, {
```
Comment (bigint) [inline packages/data/verify.js:30]:
```suggestion
  while (true) {
    const response = await fetch(url, {
```

File: packages/data/verify.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        if (err) {
+          console.error('Error reading the verified.ts file:', err);
+          return;
+        }
+
+        const regex =
+          /export const mainnetVerified = \[[\S\s]*?mainnetLensTeamMembers,[\S\s]*?...mainnetStaffs\n];/;
+        const match = tsData.match(regex);
+        // console.log('match:', match);
+
```
Comment (bigint) [inline packages/data/verify.js:82]:
```suggestion
        const match = tsData.match(regex);

```

File: packages/data/verify.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+          const existingProfiles = new Set(
+            match[0]
+              .split('\n')
+              .map((line) => line.trim())
+              .slice(1, -3)
+          );
+
+          // console.log('existingProfiles:', existingProfiles);
+
+          const regex = /'0x([\dA-Fa-f]+)'/g;
```
Comment (bigint) [inline packages/data/verify.js:94]:
```suggestion
          );

          const regex = /'0x([\dA-Fa-f]+)'/g;
```

File: packages/data/verify.js [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+            'mainnetVerified array not found in the verified.ts file.'
+          );
+        }
+      });
+    } catch (error) {
+      console.error('Error:', error);
+    }
+
+    // console.log(`Total number of IDs: ${allResponses.length}`);
+  } else {
```
Comment (bigint) [inline packages/data/verify.js:140]:
```suggestion
    }
  } else {
```

File: packages/data/verify.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+      // Load the verified.ts file
+      fs.readFile('./verified.ts', 'utf8', (err, tsData) => {
+        if (err) {
+          console.error('Error reading the verified.ts file:', err);
+          return;
+        }
+
+        const regex =
+          /export const mainnetVerified = \[[\S\s]*?mainnetLensTeamMembers,[\S\s]*?...mainnetStaffs\n];/;
+        const match = tsData.match(regex);
```
Comment (bigint) [inline packages/data/verify.js:79]: I'm mean mainnetLensTeamMembers and mainnetStaffs is not needed

**Pull request closed without merging**